src/Kiota.Builder/Validation/OpenApiSchemaComparer.cs

```diff
@@ -143,17 +144,17 @@ public int GetHashCode([DisallowNull] OpenApiDiscriminator obj)
         return hash.ToHashCode();
     }
 }
-internal class OpenApiAnyComparer : IEqualityComparer<IOpenApiAny>
+internal class JsonNodeComparer : IEqualityComparer<JsonNode>
 {
     /// <inheritdoc/>
-    public bool Equals(IOpenApiAny? x, IOpenApiAny? y)
+    public bool Equals(JsonNode? x, JsonNode? y)
     {
         if (x is null || y is null) return object.Equals(x, y);
         // TODO: Can we use the OpenAPI.NET implementation of Equals?
-        return x.AnyType == y.AnyType && string.Equals(x.ToString(), y.ToString(), StringComparison.OrdinalIgnoreCase);
+        return x.GetValueKind() == y.GetValueKind() && string.Equals(x.GetValue<string>(), y.GetValue<string>(), StringComparison.OrdinalIgnoreCase);
```

Comment:
Maybe we should be using `ToJsonString()` or `ToString()` instead of `GetValue<string>()` in this scenario?
My fear here is that it will throw if the node is not convertible to a string e.g object/array? 
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.nodes.jsonnode.getvalue?view=net-9.0#exceptions

Reply:
To have more tolerance for the difference kind of values? that's a great catch!

```suggestion
        return x.GetValueKind() == y.GetValueKind() && string.Equals(x.ToJsonString(), y.ToJsonString(), StringComparison.OrdinalIgnoreCase);
```